submissions/issue-16/milestone-1/Coelacanth/NFTStandard.cdc

```diff
@@ -0,0 +1,326 @@
+/*
+	This project is addressing problem "New Standard: NFT metadata #16"
+	The below code is untested and is meant as an illustration of the concept.
+	
+	
+	MetaDataUtil is added as a new default contract.
+	
+	Metadata is instantiated in the form of structs at the time the NFT is minted.
+	Metadata is organized by string tags, and each metadata element may have multiple tags allowing NFTs to conform to multiple tag schemas.
+	Metadata elements cannot be added after the NFT is minted, but Mutable elements can allow data to be modifiable.
+	
+	MetaDataElements are wrappers around two possible types of structs, one of which is immutable and the other of which is Mutable.
+	Immutable elements can use default types or be defined in custom structs that implement ITaggedMetaData and IImmutableMetaData. 
+		Defining a custom class allows the data elements to be immutable while allowing a developer to upgrade the contract that defines the struct to alter the tags.
+	Mutable elements store Capability pointers to IMetaDataProvider instances that can be stored anywhere.
+		Mutable elements can be used for metadata that can be modified, such as for leveling up a character.
+		Mutable elements can also be used for metadata that is shared between multiple NFTs, to minimize redundant data storage.
+			A Side-effect (benefit?) here is that if the instance is stored in a contract in the developers account, the developer will need to cover the metaData storage costs instead of the NFT holder.
+			PNG_RemoteDefaultImage is a default implementation that can be used to give an NFT a shared and/or updatable image.
+				RemotePNGProvider is an example of an implementation of IMetaDataProvider that allows multiple NFTs to share the same image via Capability reference.
+*/
+
+pub contract MetaDataUtil {
```

Comment:
In the Cadence code that Dapper has written, we tend to use descriptive names that provide context and/or indicate intent, rather than well-known prefixes/suffixes. We describe the scope, constraint, or affordance provided by Capability interfaces, e.g. `CollectionPublic`, `Provider`, `Receiver`. And for concrete types we tend to emphasize what they are or what they do, e.g. `NonFungibleToken`, `FlowEpoch`, `NodeDelegator`.

None if this is a criticism of other approaches to naming, or to the specific naming used here - which is very clear, regular, and logical, all of which is of advantage when reading a codebase. I would however imagine that we will be looking for naming that reads similarly to the existing standard and core contracts, and this may be worth considering.

Reply:
Gotcha. I will consider renaming. This is the first thing I've ever written in Cadence, so I am not very familiar with the standards as of yet.